### PR TITLE
[FIL-78] CronActor

### PR DESCRIPTION
**Pull request by artyom-yurin**
Signed-off-by: artyom-yurin <artem_yrin@mail.ru>

### Description of the Change
Cron Actor implementation

### Benefits
CronActor can be used in another structures

### Possible Drawbacks 
Lotus and Spec have different information
I use Lotus [implementation](https://github.com/filecoin-project/lotus/blob/master/chain/actors/actor_cron.go)

### Usage Examples or Tests
`test/core/vm/actor/cron_actor_test.cpp`

### CODE REVIEW

File: core/vm/actor/cron_actor.hpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace fc::vm::actor {
+
+  struct CronTableEntry {
+    Address to_addr;
+    uint64_t method_num;
+  };
+
+  struct CronActor {
+    // Entries is a set of actors (and corresponding methods) to call during
```
Comment (Alexey-N-Chernyshov) [inline core/vm/actor/cron_actor.hpp]: Please, use `/** */` for multiline comments

Comment (Alexey-N-Chernyshov) [reply core/vm/actor/cron_actor.hpp]: Don't copy spec when it is not needed. 
`make it easy to add a handler to Cron elsewhere in the spec code`

File: core/vm/actor/cron_actor.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+fc::outcome::result<void> fc::vm::actor::CronActor::EpochTick(
+    fc::vm::actor::Actor &actor,
+    fc::vm::VMContext &vmctx,
+    const std::vector<uint8_t> &params) {
+  if (!(vmctx.message().from == kCronAddress)) {
+    return CronActorError::WRONG_CALL;
+  }
+
+  for (const auto& entry : entries) {
+    OUTCOME_TRY(vmctx.send(entry.to_addr, entry.method_num, BigInt(1), {}));
```
Comment (Alexey-N-Chernyshov) [inline core/vm/actor/cron_actor.cpp]: `BigInt(1)` - is it ok to send 1 token every tick?

File: test/core/vm/CMakeLists.txt [hunk 3]
```diff
@@ -3,6 +3,10 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}
+        ${PROJECT_SOURCE_DIR}/core
```
Comment (Alexey-N-Chernyshov) [inline test/core/vm/CMakeLists.txt]: Not sure I understand it. Why do you include /core into test/core/vm?

Comment (artyom-yurin) [reply test/core/vm/CMakeLists.txt]: This is necessary to include VMContext Mock

File: core/vm/actor/cron_actor.cpp (hunk 2 above)
Comment (artyom-yurin) [reply core/vm/actor/cron_actor.cpp]: I saw it in Lotus
But there they sent 0 instead of 1. I will fix it

Comment (artyom-yurin) [reply core/vm/actor/cron_actor.cpp]: Fixed

File: core/vm/actor/cron_actor.hpp (hunk 1 above)
Comment (artyom-yurin) [reply core/vm/actor/cron_actor.hpp]: Fixed

File: test/core/vm/CMakeLists.txt (hunk 3 above)
Comment (Alexey-N-Chernyshov) [reply test/core/vm/CMakeLists.txt]: Please, left TODO to remove it after VMContext is implemented.

**Pull request merged**